shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeIngressServiceImpl.java

```diff
@@ -79,6 +67,42 @@ public Deployment fetchOrCreateBridgeIngressDeployment(BridgeIngress bridgeIngre
         return kubernetesClient.apps().deployments().inNamespace(bridgeIngress.getMetadata().getNamespace()).create(deployment);
     }
 
+    @Override
+    public Service fetchOrCreateBridgeIngressService(BridgeIngress bridgeIngress, Deployment deployment) {
+        Service service = kubernetesClient.services().inNamespace(bridgeIngress.getMetadata().getNamespace()).withName(bridgeIngress.getMetadata().getName()).get();
+
+        if (service != null) {
+            return service;
+        }
```

Comment:
Is this thread safe? is it possible that two concurrent requests will produce two instances of the service?
If the second call is going to fail we might need to add a try/catch at [line 85](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/101/files#diff-93fb913d2b2f15cb9a101f0d12c29a300b6dbe266ba067417c23da2ef69189d2R85) and in that case execute again the call to load the existing service.
Wdyt?
Btw what about at least add a log information if `service == null` to log the fact that a new service instance is going to be created? This can help to debug similar issues

Reply:
Even if this `fetchOrCreateBridgeIngressService` gets created concurrently it would not be a problem: the line 85 can't create two instances of the service because it is not possible to have two services with the same resource name (in the same namespace). 

Instead, I smell there could be a concurrency problem in the BridgeIngressController itself. But honestly I've never seen any concurrency management in any operator controller so far: let me double check with the java operator sdk team

Reply:
> Even if this `fetchOrCreateBridgeIngressService` gets called concurrently it would not be a problem: the line 85 can't create two instances of the service because it is not possible to have two services with the same resource name (in the same namespace).
> 
> Instead, I smell there could be a concurrency problem in the BridgeIngressController itself. But honestly I've never seen any concurrency management in any operator controller so far: let me double check with the java operator sdk team

Ok it is possible that the operator processes events sequentially so there is no concurrency problem 👍 
I'm asking because the code by itself is not preventing unexpected/undesired scenarios

Reply:
Checked with the java sdk team: the concurrency is managed by the java sdk so we don't have to worry about it.

Reply:
Checked with the java sdk team: the concurrency is managed by the sdk itself so we don't have to worry about it